iphone/Maps/UI/Help/AboutController/AboutController.swift

```diff
@@ -0,0 +1,535 @@
+    ])
+    donationView.widthAnchor.constraint(equalTo: stackView.widthAnchor).isActive = isDonateEnabled()
+
+    view.layoutIfNeeded()
+    updateCollection()
+  }
+
+  func updateCollection() {
+    socialMediaCollectionView.collectionViewLayout.invalidateLayout()
+    DispatchQueue.main.async {
+      self.socialMediaCollectionViewHeighConstraint.constant = self.socialMediaCollectionView.collectionViewLayout.collectionViewContentSize.height
+    }
+  }
+
+  func isDonateEnabled() -> Bool {
+    return Settings.donateUrl() != nil
+  }
+
+  func buildInfoTableViewData() -> [AboutInfoTableViewCellModel] {
+    let infoContent: [AboutInfo] = [.faq, .reportMapDataProblem, .reportABug, .news, .volunteer, .rateTheApp]
+    let data = infoContent.map { [weak self] aboutInfo in
+      return AboutInfoTableViewCellModel(title: aboutInfo.title, image: aboutInfo.image, didTapHandler: {
+        switch aboutInfo {
+        case .faq:
+          self?.navigationController?.pushViewController(FaqController(), animated: true)
+        case .reportABug:
+          guard let link = aboutInfo.link else { return }
+          self?.sendEmailWith(header: "Organic Maps Bugreport", toRecipients: [link])
+        case .reportMapDataProblem, .volunteer, .news:
+          self?.openUrl(aboutInfo.link)
+        case .rateTheApp:
+          UIApplication.shared.rateApp()
+        default:
+          break
+        }
+      })
+    }
+    return data
+  }
+
+  func buildSocialMediaCollectionViewData() -> [SocialMediaCollectionViewCellModel] {
+    let socialMediaContent: [SocialMedia] = [.telegram, .github, .instagram, .twitter, .linkedin, .organicMapsEmail, .reddit, .matrix, .facebook, .fosstodon]
+    let data = socialMediaContent.map { [weak self] socialMedia in
+      return SocialMediaCollectionViewCellModel(image: socialMedia.image, didTapHandler: {
+        switch socialMedia {
+        case .telegram: fallthrough
+        case .github: fallthrough
+        case .reddit: fallthrough
+        case .matrix: fallthrough
+        case .fosstodon: fallthrough
+        case .facebook: fallthrough
+        case .twitter: fallthrough
+        case .instagram: fallthrough
+        case .linkedin:
+          self?.openUrl(socialMedia.link, inSafari: true)
+        case .organicMapsEmail:
+          guard let link = socialMedia.link else { return }
+          self?.sendEmailWith(header: "Organic Maps", toRecipients: [link])
+        }
+      })
+    }
+    return data
+  }
+
+  // Returns a human-readable maps data version.
+  static func formattedMapsDataVersion() -> String {
+    // First, convert version code like 220131 to a date.
+    let df = DateFormatter()
+    df.locale = Locale(identifier:"en_US_POSIX")
+    df.dateFormat = "yyMMdd"
+    let mapsVersionInt = FrameworkHelper.dataVersion()
+    let mapsDate = df.date(from: String(mapsVersionInt))!
+    // Second, print the date in the local user's format.
+    df.locale = Locale.current
+    df.dateStyle = .long
```

Comment:
On the screenshot "28 February 2024" is displayed. In swift playground I seen "February 28, 2024" for .long (and it looks/reads better for native speakers). Is it because df.dateFormat was not reset above?

Reply:
Yes! It is because `Locale.current` defined not by language, but by the user region settings and may vary.
And reset previous `df.locale = Locale(identifier:"en_US_POSIX")`

Overall it is better to use the `current`, because `en_US` and `en_UK` have different standards of writing dates:
en_UK :  2 February 2024
en_US: February 2, 2024 

So there are different native speakers even for the same language and it better to use different formats.

Reply:
Aha, so you have en_UK locale set. Then its ok )